ta/pkcs11/src/pkcs11_token.c

```diff
@@ -900,6 +900,17 @@ enum pkcs11_rc entry_ck_token_initialize(uint32_t ptypes, TEE_Param *params)
 
 	update_persistent_db(token);
 
+	/* Remove all persistent objects */
+	if (token->db_objs && token->db_objs->count > 0) {
```

Comment:
minor: testing `token->db_objs->count` may not be needed. `while (!LIST_EMPTY(&token->object_list)) { ... }` should do the work:
```c
	/* Remove all persistent objects */
	while (!LIST_EMPTY(&token->object_list)) {
		struct pkcs11_object *obj = LIST_FIRST(&token->object_list);

		unregister_persistent_object(token, obj->uuid);
		cleanup_persistent_object(obj, token);
	}
```

Reply:
You are right. I'll push an update. Thanks